p2p/server/deadline_adjuster.go

```diff
@@ -166,3 +166,11 @@ func (dadj *deadlineAdjuster) Write(p []byte) (n int, err error) {
 	}
 	return n, nil
 }
+
+// ReadByte implements io.ByteReader, which is needed for varint.ReadUvarint, which is
+// used to read request length.
+func (dadj *deadlineAdjuster) ReadByte() (byte, error) {
+	var b [1]byte
+	_, err := io.ReadFull(dadj, b[:])
+	return b[0], err
+}
```

Comment:
Why is `io.ReadFull` necessary? Doesn't `Read` guarantee that at least 1 byte is read if `err == nil`?

```suggestion
func (dadj *deadlineAdjuster) ReadByte() (byte, error) {
	var b [1]byte
	_, err := dadj.Read(b[:])
	return b[0], err
}
```

Reply:
According to [io.Reader docs](https://pkg.go.dev/io#Reader):

> Implementations of Read are discouraged from returning a zero byte count with a nil error, except when len(p) == 0. Callers should treat a return of 0 and nil as indicating that nothing happened; in particular it does not indicate EOF. 

Thus, while it's not good for implementations to return zero byte count with a nil error, this possibility cannot be ruled out completely, and the callers are told explicitly to be able to handle such situations; which `io.ReadFull` does.